Bottom: c1ada38a3a7858a58cafdcc2e68b25dfd1a4b3da
Top:    fc06a8968b7fb49657ca8d0fc1f7a87243cd4d54
Author: Jason Wang <jasowang@redhat.com>
Date:   2018-01-22 10:25:52 +0800

tun: slient sparse warning about missing rcu annotations

This patch fixes the following sparse warnings:



Signed-off-by: Jason Wang <jasowang@redhat.com>


---

diff --git a/drivers/net/tun.c b/drivers/net/tun.c
index 6988746..2bc18b1 100644
--- a/drivers/net/tun.c
+++ b/drivers/net/tun.c
@@ -2225,7 +2225,8 @@ static void tun_prog_free(struct rcu_head *rcu)
 	kfree(prog);
 }
 
-static int __tun_set_ebpf(struct tun_struct *tun, struct tun_prog **prog_p,
+static int __tun_set_ebpf(struct tun_struct *tun,
+			  struct tun_prog __rcu **prog_p,
 			  struct bpf_prog *prog)
 {
 	struct tun_prog *old, *new = NULL;
